src/domManipulation.ts

```diff
@@ -150,51 +186,72 @@ export function loadWeatherForecast(location, fetchResults) {
         } else if (i === 2) {
           const tempList = forecastItemInfo.getElementsByTagName("div");
           tempList[0].innerText = `${minTempResult} °C`;
-          tempList[0].style.backgroundColor = temperatureColourConverter(minTempResult);
+          tempList[0].style.backgroundColor = temperatureColourConverter(Number(minTempResult));
           tempList[1].innerText = `${maxTempResult} °C`;
-          tempList[1].style.backgroundColor = temperatureColourConverter(maxTempResult);
+          tempList[1].style.backgroundColor = temperatureColourConverter(Number(maxTempResult));
         }
       }
     }
   }
 }
 
 // Controls what happens when the network call has been unsuccessful.
-export function failedForecast(location) {
+export function failedForecast(location: string) {
   console.error(`Unable to fetch the forecast for the ${location} location`);
-  document.getElementById("popup-text").innerText = "Unable to fetch forecast";
+  const popupTextElement = document.getElementById("popup-text");
+  if (popupTextElement !== null) {
+    popupTextElement.innerText = "Unable to fetch forecast";
+  }
+  
 }
 
 // Hides the loader after the network call has been completed.
-export function hideForecastLoader(location) {
+export function hideForecastLoader(location: string) {
   if (location === "custom") {
-    document.getElementById("popup-loading-container").style.visibility = "hiddem";
-    document.getElementById("popup-loading-container").style.display = "none";
-    document.getElementById("popup-forecast-container").style.visibility = "visible";
-    document.getElementById("popup-forecast-container").style.display = "flex";
+
+    const popupLoader = document.getElementById("popup-loading-container");
+    if (popupLoader !== null) {
+      popupLoader.style.visibility = "hidden";
+      popupLoader.style.display = "none";
+    }
+   
+    const popupForecast =  document.getElementById("popup-forecast-container");
+
+    if (popupForecast !== null) {
+      popupForecast.style.visibility = "visible";
+      popupForecast.style.display = "flex";
+    }
+   
```

Comment:
one if statement, look at guard clauses, I've also seen this code snippet before, if you reuse code maybe consider putting it into its own function that you can just call instead with parameters (this is just a suggestion)

Reply:
The repetition you are referring to does two different things, this particular function hides the loader for the popup while the results are being fetched. 

The similar code above (in `displayForecastLoader`) displays the loader when content is being fetched.

Hence, the difference between the code (although it is very similar) is that the one section displays the loader and hides the forecast content for the popup, and the second section displays the forecast content for the popup and hides the loader.